### PR TITLE
Marathon Storage Tool Initial Version

### PR DESCRIPTION
**Pull request by timcharper**
This implements an initial set of features for the Marathon Storage Tool:

- list pods, apps, instances, deployments (some filtering)
- purge
- built-in help
- version compatibility assertion
- build and deploy docker image

Support for 1.5.x is available, also. We symlink the majority of the files and use bindings.sc to bridge the compatibility gap between the two versions of Marathon.

### CODE REVIEW

**Comment by timcharper** [comment]
There aren't any integration tests right now 😨 we should maybe add one or two before we merge

File: ammonite-docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM openjdk:8-jdk-alpine
```
Comment (jeschkies) [inline ammonite-docker/Dockerfile]: I'd be cool to switch the Marathon Docker image to Alpine as well :).

File: notes.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+curl -L -o /usr/local/bin/amm-2.11 https://git.io/v7cxd && sudo chmod +x /usr/local/bin/amm-2.11
```
Comment (jeschkies) [inline notes.md]: What is this file?

File: src/1.4.x/lib/bindings.sc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+}
+class MarathonStorage(args: List[String] = MarathonStorage.argsFromEnv) {
+  import helpers.Helpers._
+  implicit val actorSystem = ActorSystem()
+  implicit val actorMaterializer = ActorMaterializer()
+  implicit val scheduler: Scheduler = actorSystem.scheduler
+  implicit val timeout = Timeout(5.seconds)
+  implicit val executionContext = scala.concurrent.ExecutionContext.global
+
+  private class ScallopStub[A](name: String, value: Option[A]) extends ScallopOption[A](name) {
```
Comment (jeschkies) [inline src/1.4.x/lib/bindings.sc]: What is `ScallopStub` doing?

File: src/1.5.x/lib/dsl.sc [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+  }
+
+  def purge[T](values: Seq[T])(implicit purgeStrategy: PurgeStrategy[T], formatter: StringFormatter[T]): Unit = {
+    println()
+    println(s"Are you sure you wish to purge the following ${purgeStrategy.purgeDescription}?")
+    println()
+    val formattedValues = values.map(formatter)
+    formattedValues.foreach { v => println(s"  ${v}") }
+    println()
+    setConfirmation(formattedValues.hashCode) {
```
Comment (jeschkies) [inline src/1.5.x/lib/dsl.sc]: I just wanted to propose this and then remembered that I've seen it in your CLIs :).

File: src/1.5.x/lib/dsl.sc [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  def purge[T](queryResult: QueryResult[T])(implicit purgeStrategy: PurgeStrategy[T], formatter: StringFormatter[T]): Unit = {
+    purge(queryResult.values)
+  }
+
+  def purge[T](value: T)(implicit purgeStrategy: PurgeStrategy[T], formatter: StringFormatter[T]): Unit = {
+    purge(Seq(value))
+  }
+
+  def help: Unit = {
+    println(s"""
```
Comment (jeschkies) [inline src/1.5.x/lib/dsl.sc]: Would be nice to have a command builder generating this.

File: ammonite-docker/Dockerfile (hunk 1 above)
Comment (timcharper) [reply ammonite-docker/Dockerfile]: Unfortunately, the native mesos lib depends on glibc and does not work with Alpine's libc :( I'm not sure if it is possible to make it work.

File: notes.md (hunk 2 above)
Comment (timcharper) [reply notes.md]: Ammonite executable; we download and install ammonite. But, specifically, ammonite for Scala 2.11

File: src/1.4.x/lib/bindings.sc (hunk 3 above)
Comment (timcharper) [reply src/1.4.x/lib/bindings.sc]: We use it to disallow the configuration of storeCache and versionCacheEnabled, both of which we set to false.

File: src/1.5.x/lib/dsl.sc (hunk 5 above)
Comment (timcharper) [reply src/1.5.x/lib/dsl.sc]: Totally agreed :) I was about to, but then realized that a string would suffice for now and take much less effort :) At a future date?

File: BUILDING.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# Building
+
+Right now, only 1.4.x versions of Marathon are supported.
+
+Marathon storage tool must be linked against a specific version of Marathon.
+
+For convenience, use the Makefile!:
+
+```
+make targets/1.4.5/docker-built
```
Comment (elezar) [inline BUILDING.md]: built -> build

**Pull request merged**